repo-agent/pkg/llm/command_expansion_test.go

```diff
@@ -0,0 +1,79 @@
+package llm
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestExpandCommands(t *testing.T) {
+	// Setup temporary .gemini directory
+	tempDir := t.TempDir()
+	geminiDir := filepath.Join(tempDir, ".gemini")
+	commandsDir := filepath.Join(geminiDir, "commands")
+
+	// Create namespace directory
+	nsDir := filepath.Join(commandsDir, "google-internal")
+	if err := os.MkdirAll(nsDir, 0755); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create command file
+	cmdFile := filepath.Join(nsDir, "review-pr.toml")
+	cmdContent := `
+description = "Test Command"
+prompt = """
+Reviewing PR {{args}}.
+"""
+`
+	if err := os.WriteFile(cmdFile, []byte(cmdContent), 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name     string
+		prompt   string
+		expected string
+	}{
+		{
+			name:     "Basic command",
+			prompt:   `Run this: /google-internal:review-pr "123"`,
+			expected: `Run this: Reviewing PR 123.`,
+		},
+		{
```

Comment:
It's good practice to `defer cleanup()` at the beginning of the test function to ensure that the temporary directory is always removed, even if the test panics.